SRCH-507 remove support for Instagram

MothOnMars opened the pull request:
## Summary
This PR removes support for Instagram, as that integration hasn't worked since 2017. It also removes the code and configuration for Capistrano, which hasn't been used since at least 2016.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:
 
#### Functionality Checks
 
- [x] Code is functional, as tested locally.

- [x] Tests have been added or updated to cover the proposed changes. If not, please explain why tests aren't needed: Changes are all subtractive.
 
- [x] Automated checks pass, if applicable. If CodeClimate checks do not pass, explain reason for failures: The single failure is for a pre-existing issue. Please "approve" via the Code Climate UI.
 
- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock.

- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] Your target branch is `master`, `main`, or `production`, or you have specified the reason for an alternate branch here:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] You have squashed your commits into a single commit (exceptions: your PR includes commits with formatting-only changes, such as required by Rubocop or Cookstyle, or if this is a feature branch that includes multiple commits)
 
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

jamesmadhoc reviewed (approved):
LGTM.

I do have some comments and questions, though; hope this is an
appropriate place to ask them:

1 This ticket looks like a good preview of the one I think you sent
  my way: SRCH-1574 ("Agencies API Removal").

2 How come the Jira ticket says this needs to be done *after*
  SRCH-508 ("delete any Instagram profiles and photos from the ASIS
  index")?

3 The change to fetch_xml() (in app/workers/mrss_photos_importer.rb)
  to use the Faraday rather than the HTTP gem was done earlier in
  SRCH-2958 ("the MrssPhotosImporter should work").  (Ditto changes
  to its spec.)  I just thought this was curious, and was wondering
  if it would cause a spurious merge conflict.

4 Why was Capistrano removed?

5 Why was the space character removed in photo ID strings in
  spec/models/flickr_album_detector_spec.rb (and in
  spec/models/mrss_album_detector_spec.rb)?

MothOnMars commented:
@jamesmadhoc , the PR is a fine place to ask such questions. Answers below.

> 1 This ticket looks like a good preview of the one I think you sent my way: SRCH-1574 ("Agencies API Removal").

Yes, although the Agencies API removal should be far simpler, as it's unlikely to require additional code changes beyond removing the files.

> 2 How come the Jira ticket says this needs to be done _after_ SRCH-508 ("delete any Instagram profiles and photos from the ASIS index")?

Good catch - that should actually be switched, as this ticket should be done before that ticket. I've updated the link in Jira.

> 3 The change to fetch_xml() (in app/workers/mrss_photos_importer.rb) to use the Faraday rather than the HTTP gem was done earlier in SRCH-2958 ("the MrssPhotosImporter should work"). (Ditto changes to its spec.) I just thought this was curious, and was wondering if it would cause a spurious merge conflict.

Could you clarify where you're seeing that, other than in the PR for 2958? That file isn't part of this PR.

> 4 Why was Capistrano removed?

See my comments in the PR summary. I removed it as part of these changes, as I would otherwise have had to make changes in the Capistrano files to support the Instagram removal.

> 5 Why was the space character removed in photo ID strings in spec/models/flickr_album_detector_spec.rb (and in spec/models/mrss_album_detector_spec.rb)?

Same comment as re. `app/workers/mrss_photos_importer.rb`. Those files are not part of this PR, so I'm not sure where you're seeing those changes...? The spaces were removed and explained in a separate PR: https://github.com/GSA/asis/pull/133
